packages/google_maps_flutter/google_maps_flutter_ios/ios/Classes/FLTGoogleMapTileOverlayController.m

```diff
@@ -35,33 +38,22 @@ - (void)clearTileCache {
   [self.layer clearTileCache];
 }
 
-- (void)setFadeIn:(BOOL)fadeIn {
-  self.layer.fadeIn = fadeIn;
-}
-
-- (void)setTransparency:(float)transparency {
-  float opacity = 1.0 - transparency;
-  self.layer.opacity = opacity;
-}
-
-- (void)setVisible:(BOOL)visible {
-  self.layer.map = visible ? self.mapView : nil;
+- (void)updateFromPlatformTileOverlay:(FGMPlatformTileOverlay *)overlay {
+  [FLTGoogleMapTileOverlayController updateTileLayer:self.layer
+                             fromPlatformTileOverlay:overlay
+                                         withMapView:self.mapView];
 }
 
-- (void)setZIndex:(int)zIndex {
-  self.layer.zIndex = zIndex;
-}
++ (void)updateTileLayer:(GMSTileLayer *)tileLayer
+    fromPlatformTileOverlay:(FGMPlatformTileOverlay *)platformOverlay
+                withMapView:(GMSMapView *)mapView {
+  tileLayer.opacity = 1.0 - platformOverlay.transparency;
+  tileLayer.zIndex = (int)platformOverlay.zIndex;
+  tileLayer.fadeIn = platformOverlay.fadeIn;
+  tileLayer.tileSize = platformOverlay.tileSize;
 
-- (void)setTileSize:(NSInteger)tileSize {
-  self.layer.tileSize = tileSize;
-}
-
-- (void)updateFromPlatformTileOverlay:(FGMPlatformTileOverlay *)overlay {
-  [self setVisible:overlay.visible];
-  [self setTransparency:overlay.transparency];
-  [self setZIndex:(int)overlay.zIndex];
-  [self setFadeIn:overlay.fadeIn];
-  [self setTileSize:overlay.tileSize];
+  // This must be done last, to avoid visual flickers of default property values.
+  tileLayer.map = platformOverlay.visible ? mapView : nil;
```

Comment:
This is the actual fix; about half the objects were setting visibility somewhere other than the end.